app/src/test/java/de/westnordost/streetcomplete/quests/LastPickedValuesStoreTest.kt

```diff
@@ -7,14 +7,14 @@ import org.junit.Test
 
 class LastPickedValuesStoreTest {
 
-    @Test fun `mostCommonWithin sorts by frequency first, then recency`() {
+    @Test fun `mostCommonWithin sorts by recency`() {
         val items = sequenceOf(A, C, B, B, C, D)
-        assertEquals(items.mostCommonWithin(4, 99).toList(), listOf(C, B, A, D))
+        assertEquals(items.mostCommonWithin(4, 99).toList(), listOf(A, C, B, D))
     }
 
```

Comment:
I think a new test should be added here:

```suggestion

@Test fun `mostCommonWithin includes the most common items in the original order`() {
    val items = sequenceOf(A, A, B, B, B, C)
    assertEquals(items.mostCommonWithin(2, 99).toList(), listOf(A, B))
}

```

Reply:
It's not really covered, because in the `sorts by recency` test, all items are returned. I think the intent comes better across in my suggestion.

Merging some of the tests is not a bad idea, I guess.

Reply:
Okay, done.

As a consequence, the tests are now much more readable. I almost considered replacing the comment above `mostCommonWithin`with "see the tests for this function" since the assertion messages read like almost a direct copy-paste.